graphql-server/pkg/errors/errors.go

```diff
@@ -1,73 +1,46 @@
 package errors
 
 import (
-	"errors"
-	"fmt"
-)
+	"context"
 
-var (
-	ErrInternalError     = errors.New("Internal error")
-	ErrInvalidNodeID     = errors.New("Invalid node ID")
-	ErrValidationFailure = errors.New("Failed to validate values")
-	ErrNotFound          = errors.New("Not found")
-	ErrInvalidCursor     = errors.New("Invalid cursor")
-	ErrUnauthenticated   = errors.New("Unauthenticated")
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/vektah/gqlparser/v2/gqlerror"
 )
 
-type ServerError struct {
-	parent  error
-	message string
-}
-
-type Causer interface {
-	Cause() error
-}
-
-func (e *ServerError) Error() string {
-	return fmt.Sprintf("%s: %s", e.message, e.parent.Error())
-}
-
-func (e *ServerError) Cause() error {
-	return e.parent
-}
-
-func (e *ServerError) Is(target error) bool {
-	if e == target {
-		return true
-	}
-
-	w := e.Cause()
-	for {
-		if w == target {
-			return true
-		}
+type ServerErrorCode string
+
+const (
+	InternalError     ServerErrorCode = "INTERNAL_ERROR"
```

Comment:
should we follow apollo standard codes? https://www.apollographql.com/docs/apollo-server/data/errors/